File: cpp/Hash Maps and Sets/longest_chain_of_consecutive_numbers_brute_force.cpp
#include <vector>        
#include <algorithm>    

int longestChainOfConsecutiveNumbersBruteForce(std::vector<int>& nums) {
    if (nums.empty()) {
        return 0;
    }
    int longestChain = 0;  
    // Look for chains of consecutive numbers that start from each number.
    for (int num : nums) {
        int currentNum = num;         
        int currentChain = 1;         
        // Continue to find the next consecutive numbers in the chain.
        while (std::find(nums.begin(), nums.end(), currentNum + 1) != nums.end()) {
            currentNum += 1;
            currentChain += 1;
        }
        longestChain = std::max(longestChain, currentChain);
    }
    return longestChain;
}

File: go/Hash Maps and Sets/longest_chain_of_consecutive_numbers_brute_force.go
package hashmapsandsets

func LongestChainOfConsecutiveNumbersBruteForce(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	longestChain := 0
	// Look for chains of consecutive numbers that start from each number.
	for _, num := range nums {
		currentNum := num
		currentChain := 1
		// Continue to find the next consecutive numbers in the chain.
		for contains(nums, currentNum+1) {
			currentNum++
			currentChain++
		}
		if currentChain > longestChain {
			longestChain = currentChain
		}
	}
	return longestChain
}

// In the Python code, the while loop checks (current_num +1) in nums.
// So the helper function 'contains' is needed.
func contains(nums []int, target int) bool {
	for _, n := range nums {
		if n == target {
			return true
		}
	}
	return false
}
